llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -582,7 +668,17 @@ PreservedAnalyses SpecConstantsPass::run(Module &M,
           BitCastInst *BitCast = new BitCastInst(
               GEP, PointerType::get(SCTy, GEP->getAddressSpace()), "bc", CI);
 
-          Replacement = new LoadInst(SCTy, BitCast, "load", CI);
+          Instruction *Inst = new LoadInst(SCTy, BitCast, "load", CI);
+          Replacement = Inst;
+
+          Value *GlobVar = (IsComposite ? CI->getOperand(2) : CI->getOperand(1))
+                               ->stripPointerCasts();
+          Value *DefaultValue =
+              cast<GlobalVariable>(GlobVar)->getInitializer()->getOperand(0);
```

Comment:
Since we are using this in both native and emulation modes, it probably makes sense to outline this into a common code somewhere